src/backend/distributed/operations/shard_split.c

```diff
@@ -193,6 +228,15 @@ ErrorIfCannotSplitShardExtended(SplitOperation splitOperation,
 							   SplitTargetName[splitOperation])));
 	}
 
+	if (extern_IsColumnarTableAmTable(shardIntervalToSplit->relationId))
```

Comment:
This check will be removed / made specific for Non-Blocking only (right now for both).

Reply:
If BlockingSplit is not supported for columnar based on the above code, why should it be allowed for NonBlocking? Anyways

Reply:
I think we can relax this restriction now